---
title: "QC"
output: html_document
author: James Kondilios
---

#*Adapter Removal* <br />
This program searches for and removes remnant adapter sequences from High-Throughput Sequencing (HTS) data and (optionally) trims low quality bases from the 3' end of reads following adapter removal. AdapterRemoval can analyze both single end and paired end data, and can be used to merge overlapping paired-ended reads into (longer) consensus sequences. Additionally, the AdapterRemoval may be used to recover a consensus adapter sequence for paired-ended data, for which this information is not available.
```{bash AdapterRemoval, eval = FALSE}
# Paired-end reads:
AdapterRemoval --file1 reads_1.fq --file2 reads_2.fq --basename output_paired --trimns --trimqualities --collapse

```


#*FastQC* <br />
provide an overview of basic quality control metrics for raw next generation sequencing data. There are a number different analyses (called modules) that may be performed on a sequence data set. 
Simple information about input FASTQ file: its name, type of quality score encoding, total number of reads, read length and GC content.

```{bash FastQC, eval=FALSE}
fastqc ./good_sequence_short.fastqc
```

